pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,775 @@
+#
+#   Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+""" This module defines functions for handling Deephaven date/time data. """
+
+import jpy
+
+from deephaven2 import DHError
+from deephaven2.constants import TimeZone
+from deephaven2.dtypes import DateTime, Period
+
+SECOND = 1000000000  #: One second in nanoseconds.
+MINUTE = 60 * SECOND  #: One minute in nanoseconds.
+HOUR = 60 * MINUTE  #: One hour in nanoseconds.
+DAY = 24 * HOUR  #: One day in nanoseconds.
+WEEK = 7 * DAY  #: One week in nanoseconds.
+YEAR = 52 * WEEK  #: One year in nanoseconds.
+
+_JDateTimeUtils = jpy.get_type("io.deephaven.time.DateTimeUtils")
+
+
+def to_datetime(s: str, quiet: bool = False):
+    """ Converts a datetime string to a DateTime object
+
+    Args:
+        s (str): in the form yyyy-MM-ddThh:mm:ss and with optional sub-seconds after an
+            optional decimal point, followed by a mandatory time zone character code
+        quiet (bool, optional): when True, if the datetime string can't be parsed, this function returns None, otherwise
+            it raises an exception; default is False
+
+    Returns:
+        a DateTime
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertDateTimeQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertDateTime(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def to_period(s: str, quiet: bool = False) -> Period:
+    """ Converts a period string into a Period object.
+
+    Args:
+        s (str): a string in the form of nYnMnWnDTnHnMnS, with n being numeric values, e.g. 1W for one week, T1M for
+            one minute, 1WT1H for one week plus one hour
+        quiet (bool, optional): when True, if the period string can't be parsed, this function returns None, otherwise
+            it raises an exception; default is False
+
+    Returns:
+        Period
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertPeriodQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertPeriod(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def to_nanos(s, quiet: bool = False) -> int:
+    """ Converts a time string to nanoseconds from the Epoch.
+
+    Args:
+        s (str): in the format of: hh:mm:ss[.nnnnnnnnn]
+        quiet (bool): to return None or raise an exception if the string can't be parsed, default is False
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertTimeQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertTime(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def now():
+    """ Provides the current date/time.
```

Comment:
```suggestion
    """ Provides the current datetime.
```
As it is written, it looks like it returns one or the other.